src/dynamic_loader.cpp

```diff
@@ -26,18 +26,37 @@
 #include <migraphx/file_buffer.hpp>
 #include <migraphx/tmp_dir.hpp>
 #include <utility>
-
 #include <dlfcn.h>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
+struct dlclose_wrapper
+{
+    void operator()(void* handle)
+    {
+        if(handle)
+        {
+            dlclose(handle);
+        }
+    }
+};
+
+void check_load_error(bool flush = false)
+{
+    char* error_msg = dlerror();
+    if(not flush and error_msg)
+        MIGRAPHX_THROW("Dynamic loading or symbol lookup failed with " + std::string(error_msg));
+}
+
 struct dynamic_loader_impl
 {
     dynamic_loader_impl() = default;
+
     dynamic_loader_impl(const fs::path& p, std::shared_ptr<tmp_dir> t = nullptr)
-        : handle(dlopen(p.string().c_str(), RTLD_LAZY), &dlclose), temp(std::move(t))
+        : handle(dlopen(p.string().c_str(), RTLD_LAZY), dlclose_wrapper{}), temp(std::move(t))
```

Comment:
Instead of defining a class you could just do `manage_deleter<decltype(&dlclose), &dlclose>`.

Reply:
Done.